packages/babel-types/src/definitions/flow.js

```diff
@@ -211,6 +211,15 @@ defineType("InterfaceExtends", {
 
 defineInterfaceishType("InterfaceDeclaration");
 
+defineType("InterfaceTypeAnnotation", {
+  visitor: ["extends", "body"],
+  aliases: ["Flow", "FlowType"],
+  fields: {
+    extends: validateOptional(arrayOfType("InterfaceExtends")),
```

Comment:
If this is optional, in the generator you should check `if (node.extends && node.extends.length)` instead of `if (node.extends.length)`.